cheroot/test/test_server.py

```diff
@@ -236,3 +237,65 @@ def test_peercreds_unix_sock_with_lookup(peercreds_enabled_server):
         peercreds_text_resp = requests.get(unix_base_uri + PEERCRED_TEXTS_URI)
         peercreds_text_resp.raise_for_status()
         assert peercreds_text_resp.text == expected_textcreds
+
+
+@pytest.mark.skipif(
+    IS_WINDOWS,
+    reason=(
+        'This regression test is for a Linux bug, and the resource module '
+        'is not available on Windows.'
+    ),
+)
+def test_high_number_of_file_descriptors():
+    """Test the server does not crash with a high file-descriptor value.
+
+    This test should cause a server crash, as the server will try to
+    select() a file-descriptor higher than 1024.
+    """
+    # Imported here, as it is not available on Windows
+    import resource
+
+    # Get current resource limits to restore them later
+    (soft_limit, hard_limit) = resource.getrlimit(resource.RLIMIT_NOFILE)
+    increased_nofile_limit = 2048
+
+    # Create our server
+    httpserver = HTTPServer(
+        bind_addr=(ANY_INTERFACE_IPV4, EPHEMERAL_PORT), gateway=Gateway,
+    )
+    # Hoard a lot of file descriptors by opening and storing a lot of sockets
+    test_sockets = []
+
+    try:
+        # We have to increase the nofile limit above 1024
+        # Otherwise we see a 'Too many files open' error, instead of
+        # an error due to the file descriptor number being too high
+        resource.setrlimit(
+            resource.RLIMIT_NOFILE,
+            (increased_nofile_limit, hard_limit),
+        )
+
+        # Open a lot of file descriptors, so the next ones the server
+        # opens are higher than 1024
+        for i in range(1024):
+            test_sockets.append(socket.socket())
+        assert test_sockets[-1].fileno() >= 1024
+
+        httpserver.prepare()
+        # This will trigger a crash if select() is used in the implementation
+        httpserver.tick()
+
+        with closing(socket.socket()) as sock:
```

Comment:
Is `closing()` necessary? doesn't socket already support a CM protocol? Any corner cases?

Reply:
Ah, nevermind, that's py3-only

Reply:
So let's explicitly note the motivation:
```suggestion
        with closing(socket.socket()) as sock:  # closing is here for py2-compat
```